docs/api/configuration.md

```diff
@@ -0,0 +1,81 @@
+
+## `data_array` (Required)
+The value for this key should be a mapping specifying how to interpret the information in the target dataset. The following table describes keys available in this mapping schema.
+
+| Key | Required? | Type | Value Description |
+|-----|-----------|------|-------------------|
+|`active`|YES     |`str` |This should be a name of an array that exists in the current dataset. This value will be passed to `DatasetBuilder.set_data_array_active_name`. |
+|`x`  |NO (default=None)  |`str`|This should be the name of a coordinate that exists in the active data array. This value will be passed to `DatasetBuilder.set_data_array_axis_names`.|
+|`y`  |NO (default=None)  |`str`|This should be the name of a coordinate that exists in the active data array. This value will be passed to `DatasetBuilder.set_data_array_axis_names`.|
+|`z`  |NO (default=None)  |`str`|This should be the name of a coordinate that exists in the active data array. This value will be passed to `DatasetBuilder.set_data_array_axis_names`.|
+|`t`  |NO (default=None)  |`str`|This should be the name of a coordinate that exists in the active data array. This value will be passed to `DatasetBuilder.set_data_array_axis_names`.|
+|`t_index` |NO (default=0)|`int`|The index of the current time slice. Must be an integer >= 0 and < the length of the current time coordinate.This value will be passed to `DatasetBuilder.set_data_array_time_index`.|
+
+## `data_slices` (Optional)
+The value for this key should be a mapping of coordinate names (which are likely used as values for `x` | `y` | `z` | `t` in the `data_array` section) to slicing arrays. This mapping will be formatted and passed to `DatasetBuilder.set_data_array_coordinates`.
+
+Each slicing array should be a list of three values `[start, stop, step]`.
+
+`start`: the coordinate value at which the sliced data should start (inclusive)
+
+`stop`: the coordinate value at which the sliced data should stop (exclusive)
+
+`step`: an integer > 0 which represents the number of items to skip when slicing the data (e.g. step=2 represents 0.5 resolution)
+
+## `ui` (Optional)
+The value for this key should be a mapping of any number of UI state values. The following table describes keys available in this mapping schema.
+
+
+| Key | Required? | Type | Value Description |
+|-----|-----------|------|-------------------|
+|`main_drawer`|NO (default=True)|`bool`|If true, open the lefthand drawer for dataset and data array browsing/selection.|
+|`axis_drawer`|NO (default=False)|`bool`|If true, open the righthand drawer for axis assignment/slicing. **Note:** By default, this becomes True when an active data array is selected.|
+|`unapplied_changes`|NO (default=False)|`bool`|If true, show "Apply and Render" button, which when clicked will apply any unapplied changes and rerender.|
+|`error_message`|NO (default=None)|`str` | `None`|If not None, this string will show as the error message above the render area.|
```

Comment:
what is the purpose of the entry in the config?

Reply:
For `error_message` and `unapplied_changes`, there isn't much purpose for the standard use case, and we don't write those values when exporting the state. But the import function technically will read them, since it does a simple state update call with this dictionary. I wanted to include all possible values in the documentation in case someone would need them, but I already took out stuff like `ui_action_message` and `ui_current_time_string`. 

I can take out more if having these is confusing, but I can imagine an advanced use case would exist where someone might need to propagate a scripting error to the interface if they generate config files via code.

Reply:
I see. I would add `(Aimed to be used for advanced use case).` or something at the end of your current description. Otherwise, it can be good as-is. I was just not seeing the purpose of it.